packages/fe-fpm-writer/src/page/index.ts

```diff
@@ -1,146 +1,16 @@
-        };
-        newRoute.pattern = `${pattern.base}/${pattern.navEntity}${pattern.navKey}:?query:`;
-        if (sourceRoute?.target?.constructor === Array) {
-            const pages = sourceRoute.target;
-            // FCL only supports 3 columns, therefore, show the page in fullscreen if it is the 4th level of navigation
-            newRoute.target =
-                pages.length > 2
-                    ? [newRoute.name]
-                    : ([...pages, newRoute.name] as [string | ManifestNamespace.RouteTargetObject]);
-        } else {
-            newRoute.target = config.fcl ? [newRoute.name] : newRoute.name;
-        }
-    } else {
-        newRoute.pattern = routes.length > 0 ? `${config.entity}:?query:` : ':?query:';
-        newRoute.target = config.fcl ? [newRoute.name] : newRoute.name;
-    }
-    routes.push(newRoute);
-}
-
-/**
- * Validate the input parameters for the execution of generateCustomAction.
- *
- * @param {string} basePath - the base path
- * @param {CustomPage} config - the custom page configuration
- * @param {Editor} [fs] - the memfs editor instance
- * @returns {Promise<Editor>} the updated memfs editor instance
- */
-export function validateCustomPageConfig(basePath: string, config: CustomPage, fs?: Editor): Editor {
-    // common validators
-    validateVersion(config.ui5Version);
-    if (!fs) {
-        fs = create(createStorage());
-    }
-    validateBasePath(basePath, fs);
-
-    // validate config against the manifest
-    const manifest = fs.readJSON(join(basePath, 'webapp/manifest.json')) as Manifest;
-    if (config.navigation) {
-        if (!manifest['sap.ui5']?.routing?.targets?.[config.navigation.sourcePage]) {
-            throw new Error(`Could not find navigation source ${config.navigation.sourcePage}!`);
-        }
-        const routes: { [k: string]: ManifestNamespace.RouteWithoutName } = {};
-        if (manifest['sap.ui5']?.routing?.routes?.constructor === Array) {
-            manifest['sap.ui5'].routing.routes.forEach((routeWithName) => {
-                routes[routeWithName.name] = routeWithName;
-            });
-        } else {
-            Object.assign(routes, manifest['sap.ui5']?.routing?.routes ?? {});
-        }
-
-        const route = routes[config.navigation?.sourcePage];
-        if (!route || !route.pattern || !route.target) {
-            throw new Error(
-                `Missing or invalid routing configuration for navigation source ${config.navigation.sourcePage}!`
-            );
-        }
-    }
-
-    return fs;
-}
-
-/**
- * Add a custom page to an existing UI5 application.
- *
- * @param {string} basePath - the base path
- * @param {CustomPage} data - the custom page configuration
- * @param {Editor} [fs] - the memfs editor instance
- * @returns {Promise<Editor>} the updated memfs editor instance
- */
-export function generateCustomPage(basePath: string, data: CustomPage, fs?: Editor): Editor {
```

Comment:
as this function gets deleted, it is an incompatibility that needs to be considered when consuming the new open-ux-tools version at tools-suite - coding needs to be adapted

Reply:
The function didn't get deleted. It moved to the `custom.ts` file and is imported and exported here again. So, everything is backward compatible.

Reply:
Yes, this was clear, but the caller needs to adapt, as the function was moved and renamed. Just a reminder for us at tools-suite.

Reply:
No, the caller doesn't need to adapt. Both previously possible calls are still valid
```
import { generateCustomPage } from '@sap-ux/fe-fpm-writer';
import { generateCustomPage } from '@sap-ux/fe-fpm-writer/dist/page';
```

Reply:
OK then